flask_resources/resources.py

```diff
@@ -0,0 +1,130 @@
+from .views import ListView, ItemView, SingletonView
+
+
+ITEM_VIEW_SUFFIX = "_item_view"
+LIST_VIEW_SUFFIX = "_list_view"
+SINGLETON_VIEW_SUFFIX = "_singleton_view"
+
+
+class Resource(object):
+    """Resource controller interface."""
+
+    def __init__(self, controller, config):
+        """Initialize the base resource."""
+        self.controller = controller
+        self.config = config
+        self.bp_name = None
+
+    ## Primary interface
+    def search(self, request_context):
+        """Perform a search over the items."""
+        abort(405)
+
+    def create(self):
+        """Create an item."""
+        abort(405)
+
+    def read(self, *args, **kwargs):
+        """Read an item."""
+        abort(405)
+
+    def update(self, data, *args, **kwargs):
+        """Update an item."""
+        abort(405)
+
+    def partial_update(self, data, *args, **kwargs):
+        """Partial update an item."""
+        abort(405)
+
+    def delete(self, *args, **kwargs):
+        """Delete an item."""
+        abort(405)
+
+    ## Secondary interface
+    def as_blueprint(self, name, **bp_kwargs):
+        """Create blueprint and register rules only for the RecordResource."""
+        self.bp_name = name
+        blueprint = Blueprint(name, __name__, **bp_kwargs)
+
+        for rule in self.create_url_rules(name):
+            blueprint.add_url_rule(**rule)
+
+        return blueprint
+
+    def create_url_rules(self, bp_name):
+        return [
+            {
+                "rule": self.config.item_route,
+                "view_func": ItemView.as_view(
+                    name="{}{}".format(bp_name, ITEM_VIEW_SUFFIX),
+                    resource=self,
+                ),
+            }
+        ]
+
+    def create_error_handlers(self):
+        return []
+
+    def load_item_from_request(self):
+        # FIXME: Code default
+        # self.config.item_loader()
+        pass
+
+    def make_list_response(self, item_list, http_code):
+        # FIXME: Code default
+        # self.config.list_serializer()
+        pass
+
+    def make_item_response(self, item, http_code):
+        # FIXME: Code default
+        # self.config.item_serializer()
+        pass
+
+    def make_error_response(self, error_data, http_code):
+        # FIXME: Code default
+        # self.config.error_serializer()
+        pass
+
+
+class CollectionResource(Resource):
+    def create_url_rules(self, bp_name):
+        return [
+            {
+                "rule": self.config.item_route,
+                "view_func": ItemView.as_view(
+                    name="{}{}".format(bp_name, ITEM_VIEW_SUFFIX),
+                    resource=self,
+                ),
+            },
+            {
+                "rule": self.config.list_route,
+                "view_func": ListView.as_view(
+                    name="{}{}".format(bp_name, LIST_VIEW_SUFFIX),
+                    resource=self,
+                ),
+            },
+        ]
+
+
+class SingletonResource(Resource):
```

Comment:
This is for non items, like actions, correct?

Reply:
Correct. Though potentially it can be used by an item if its usage fits the singleton pattern. However, It is true that right now that only includes `actions`.